### PR TITLE
feat: improve E2E with meaningful roles

### PR DESCRIPTION
**Pull request by danut13**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [X] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

Before this PR, the generated Ethereum node image which contained the Pantos contracts pre-deployed was built using the old approach of the ownership model (one owner of everything).

This PR adds full support for the Pantos RBAC, with meaningful roles (gnosis safes) which have different owners.

More specifically, when building the docker image, now we additionally do the following:

1. Deploy the safe contracts, which are owned by separate accounts (now generated using the mnemonic)
2. Simulate the transactions using the foundry script
3. Process the simulated transaction to be ready for signing
4. Sign the processed simulated transactions (by the owners of the safes, which are Foundry keystores)
5. Submit the signed transactions
6. Send the PAN tokens from the super critical role safe to the gas payer account

### CODE REVIEW

File: deploy_chain.sh [hunk 1]
```diff
@@ -4,10 +4,25 @@ set -e
 
 mkdir -p .foundry/keystores
 
-echo '{"crypto":{"cipher":"aes-128-ctr","cipherparams":{"iv":"5159c4a67ae85ebbbc2ac7411cc965ef"},"ciphertext":"07a8f37ddc742b96688c6612d1d4c0f0e3a3e2902a3544ca9fe8c081aad22786","kdf":"scrypt","kdfparams":{"dklen":32,"n":8192,"p":1,"r":8,"salt":"3ef92b928e55339fb020a68358be76f7f186bdf40978a353e5f7c7e244936ac0"},"mac":"aa6a17a67b132a3f1babd4a0b02bfaa28ee4810c43b59a31780b921c8c4ac0b4"},"id":"d27236b7-d5c7-441e-aa72-1178bf76eb25","version":3}' > .foundry/keystores/local_deployer
-
 MNEMONIC="test test test test test test test test test test test junk"
 
+GAS_PAYER_PRIVATE_KEY=$(cast wallet private-key --mnemonic "$MNEMONIC" --mnemonic-index 0) # 0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266
+PAUSER_PRIVATE_KEY=$(cast wallet private-key --mnemonic "$MNEMONIC" --mnemonic-index 100) # 0x8C3229EC621644789d7F61FAa82c6d0E5F97d43D
```
Comment (chandraprakash) [inline deploy_chain.sh:10]: nice one with `--mnemonic-index`  !!

File: safe-ledger/cli/sign_with_cast_wallet.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
+import json
+import subprocess
+import sys
+import typing
+
+
+def sign_safe_tx(safe_tx_data: dict[str, typing.Any],
+                 account_name: str) -> str | None:
+    safe_tx_hex = json.dumps(safe_tx_data)
+    cmd = [
```
Comment (chandraprakash) [inline safe-ledger/cli/sign_with_cast_wallet.py:10]: once `--ledger` supports clear signing, we could switch to this approach to use ledger too:)

File: deploy_chain.sh (hunk 1 above)
Comment (danut13) [reply deploy_chain.sh:10]: Thanks!

File: safe-ledger/cli/sign_with_cast_wallet.py (hunk 2 above)
Comment (danut13) [reply safe-ledger/cli/sign_with_cast_wallet.py:48]: Yep, that would be nice

File: deploy_chain.sh [hunk 3]
```diff
@@ -40,6 +56,29 @@ DATA_DIR="$ROOT_DIR/data-static"
 
 mkdir -p $DATA_DIR
 
+# Sign the safe transactions with the given signers generated 
+# using the given script name, function name, chain, and chain_id
```
Comment (chandraprakash) [inline deploy_chain.sh:60]: informational: worth mentioning it picks the latest run of script/chain-id

File: script/SubmitSafeTxs.s.sol [hunk 4]
```diff
@@ -48,9 +49,10 @@ contract SubmitSafeTxs is Script {
 
     function run() external {
         // read transactions
+        string memory blockchainName = determineBlockchain().name;
         string memory path = string.concat(
-            vm.projectRoot(),
-            "/flat_output.json"
+            blockchainName,
+            "_flat_output.json"
```
Comment (chandraprakash) [inline script/SubmitSafeTxs.s.sol:55]: Note for us, to remove this hard-coding  `_flat_output.json` introduced by me :)

**Review by chandraprakash — DISMISSED** [review]

**Review by markuslevonyak — DISMISSED** [review]

**Review by chandraprakash — APPROVED** [review]

**Review by markuslevonyak — APPROVED** [review]

**Pull request merged**